src/plugins/integrations/AWSIntegration.ts

```diff
@@ -128,7 +130,8 @@ class AWSIntegration implements Integration {
             }
           } else if (serviceName === 'sns') {
             const operationType = SNSRequestTypes[operationName];
-            let topicName = request.params ? request.params.TopicArn : AWS_SERVICE_REQUEST;
+            let topicName = koalas(request.params.TopicArn, AWS_SERVICE_REQUEST);
```

Comment:
If topic name is not valid for the request, it is ok to use `AWS_SERVICE_REQUEST ` as the default span name, but it should not be used for `AwsSNSTags.QUEUE_NAME` tag. So you may handle this by introducing another local variable for span name.

Reply:
Done. Introduced a `spanName` variable.